gensim/models/wrappers/ldavowpalwabbit.py

```diff
@@ -0,0 +1,537 @@
+
+        corpus_size = write_corpus_as_vw(corpus, self._corpus_filename)
+
+        cmd = self._get_vw_train_command(corpus_size)
+
+        _run_vw_command(cmd)
+
+        # ensure that future updates of this model use correct offset
+        self.offset += corpus_size
+
+    def update(self, corpus):
+        """Update existing model (if any) on corpus."""
+        if not os.path.exists(self._model_filename):
+            return self.train(corpus)
+
+        LOG.debug('Updating exiting model from corpus')
+
+        # reset any existing topics generated
+        self._topics = None
+
+        corpus_size = write_corpus_as_vw(corpus, self._corpus_filename)
+
+        cmd = self._get_vw_update_command(corpus_size)
+
+        _run_vw_command(cmd)
+
+        # ensure that future updates of this model use correct offset
+        self.offset += corpus_size
+
+    def log_perplexity(self, chunk):
+        """Return per-word lower bound on log perplexity.
+
+        Also logs this and perplexity at INFO level.
+        """
+        vw_data = self._predict(chunk)[1]
+        corpus_words = sum(cnt for document in chunk for _, cnt in document)
+        bound = -vw_data['average_loss']
+        LOG.info("%.3f per-word bound, %.1f perplexity estimate based on a "
+                 "held-out corpus of %i documents with %i words",
+                 bound,
+                 numpy.exp2(-bound),
+                 vw_data['corpus_size'],
+                 corpus_words)
+        return bound
+
+    def print_topics(self, num_topics=10, num_words=10):
+        return self.show_topics(num_topics, num_words, log=True)
+
+    def show_topics(self, num_topics=10, num_words=10,
+                    log=False, formatted=True):
+        if num_topics < 0 or num_topics >= self.num_topics:
+            num_topics = self.num_topics
+        else:
+            num_topics = min(num_topics, self.num_topics)
+
+        chosen_topics = range(num_topics)
+        shown = []
+
+        for i in chosen_topics:
+            if formatted:
+                topic = self.print_topic(i, topn=num_words)
+            else:
+                topic = self.show_topic(i, topn=num_words)
+
+            shown.append(topic)
+
+            if log:
+                LOG.info("topic #%i (%.3f): %s", i, self.alpha, topic)
+
+        return shown
+
+    def print_topic(self, topicid, topn=10):
+        return ' + '.join(['{0:.3f}*{1}'.format(v[0], v[1])
+                           for v in self.show_topic(topicid, topn)])
+
+    def show_topic(self, topicid, topn=10):
+        topics = self._get_topics()
+        topic = topics[topicid]
+        bestn = numpy.argsort(topic)[::-1][:topn]
+        return [(topic[t_id], self.id2word[t_id]) for t_id in bestn]
+
+    def save(self, fname, *args, **kwargs):
+        """Serialise this model to file with given name."""
+        if os.path.exists(self._model_filename):
+            LOG.debug("Reading model bytes from '%s'", self._model_filename)
```

Comment:
Vowpal Wabbit uses its own binary model format, so when saving, this is read into a variable, so that it's serialised as part of this of this class (this means there's just a single serialised file containing both the model and the wrapper class).

The same (but opposite) goes for load(), when the wrapper is loaded, it writes the binary Vowpal Wabbit model back to disk again, so that VW can start using it right way (i.e. for predictions, further updates, etc.).

It is a bit confusing though, so I can add some comments around that.